### ISSUE TITLE
Cannot initialize plugin with Packer 1.11.z due to plugin version mismatch error

### ISSUE THREAD

**Issue by F21** [YaleUniversity/packer-plugin-goss#90]
When trying to initialize the plugin with Packer 1.11.z, I get the following error:
```
Failed getting the "github.com/YaleUniversity/goss" plugin:
11 errors occurred:
        * Continuing to next available version: binary reported version ("0.0.1") is different from the expected "3.2.10", skipping
        * Continuing to next available version: binary reported version ("0.0.1") is different from the expected "3.2.8", skipping
        * Continuing to next available version: binary reported version ("0.0.1") is different from the expected "3.2.7", skipping
        * Continuing to next available version: binary reported version ("0.0.1") is different from the expected "3.2.6", skipping
        * Continuing to next available version: binary reported version ("0.0.1") is different from the expected "3.2.5", skipping
        * Continuing to next available version: binary reported version ("0.0.1") is different from the expected "3.2.4", skipping
        * Continuing to next available version: binary reported version ("0.0.1") is different from the expected "3.2.3", skipping
        * Continuing to next available version: binary reported version ("0.0.1") is different from the expected "3.2.2", skipping
        * Continuing to next available version: binary reported version ("0.0.1") is different from the expected "3.2.1", skipping
        * could not get sha256 checksum file for github.com/YaleUniversity/goss version 3.2.0. Is the file present on the release and correctly named ? GET https://github.com/YaleUniversity/packer-plugin-goss/releases/download/v3.2.0/packer-plugin-goss_v3.2.0_SHA256SUMS: 404  []
        * could not install any compatible version of plugin "github.com/YaleUniversity/goss"
```
It seems there's an issue with setting the version number using ldflags in Goreleaser: https://github.com/YaleUniversity/packer-plugin-goss/blob/master/main.go#L14

**Comment by FalcoSuessgott** [comment]
You have to use the latest plugin version (3.2.10) in your packer plugin configuration

**Comment by F21** [comment]
I am using:
```hcl
 goss = {
      version = "~> 3.2"
      source  = "github.com/YaleUniversity/goss"
    }
```
From testing locally, I don't think `main.Version` is being set correctly in goreleaser's ldflags config.

**Comment by FalcoSuessgott** [comment]
Will check asap, something is going on I tested the release CI assets a few times but I guess one PR messed with the goreleaser config ..

**Comment by F21** [comment]
I don't think it was ever working properly before. Looking at the history of `.goreleaser.yml`, these 2 lines have never changed:
- https://github.com/YaleUniversity/packer-plugin-goss/blob/master/.goreleaser.yml#L16
- https://github.com/YaleUniversity/packer-plugin-goss/blob/master/.goreleaser.yml#L25

I think the version has always been set to `0.0.1` and packer has only started enforcing the version check in the `1.11.z` release.

**Comment by FalcoSuessgott** [comment]
Yeah seems likes it but a local packer init has always worked for me. But will investigate later and hopefully fix it for good

**Comment by F21** [comment]
The plugin version check was only released in Packer 1.11.0, which was released a few days ago.
See: 
- https://github.com/hashicorp/packer/pull/12915
- https://github.com/hashicorp/packer/blob/main/CHANGELOG.md#improvements (11th dot point)

**Comment by FalcoSuessgott** [comment]
Oh that makes sense! Thanks for investigating
